src/BlazorUI/Demo/Client/App/MainPage.xaml.cs

```diff
@@ -34,22 +67,50 @@ public MainPage()
         });
     }
 
-    private async void ContentPage_Loaded(object sender, EventArgs e)
+    private void SetupStatusBar()
     {
-        try
+        Microsoft.Maui.Handlers.WindowHandler.Mapper.AppendToMapping(nameof(IWindow), (handler, view) =>
         {
-#if WINDOWS && RELEASE
-            var webView2 = (blazorWebView.Handler.PlatformView as Microsoft.UI.Xaml.Controls.WebView2);
-            await webView2.EnsureCoreWebView2Async();
+#if ANDROID
+            var window = handler.PlatformView.Window;
+            if (window != null && Android.OS.Build.VERSION.SdkInt >= Android.OS.BuildVersionCodes.Lollipop)
+            {
+                window.ClearFlags(Android.Views.WindowManagerFlags.TranslucentStatus);
+                window.AddFlags(Android.Views.WindowManagerFlags.DrawsSystemBarBackgrounds);
+                window.SetStatusBarColor(Android.Graphics.Color.Transparent);
 
-            var settings = webView2.CoreWebView2.Settings;
-            settings.IsZoomControlEnabled = false;
-            settings.AreBrowserAcceleratorKeysEnabled = false;
+                window.DecorView.SystemUiVisibility = (Android.Views.StatusBarVisibility)(Android.Views.SystemUiFlags.LayoutFullscreen | Android.Views.SystemUiFlags.LayoutStable | Android.Views.SystemUiFlags.LightStatusBar);
+                if (AppInfo.Current.RequestedTheme == AppTheme.Dark)
+                {
+                    window.DecorView.SystemUiVisibility &= ~(Android.Views.StatusBarVisibility)Android.Views.SystemUiFlags.LightStatusBar;
+                }
+            }
 #endif
-        }
-        catch (Exception)
-        {
-            throw;
-        }
+
+#if MACCATALYST
+            var window = handler.PlatformView.WindowScene;
+            if (window != null)
+            {
+                window.Titlebar.TitleVisibility = UIKit.UITitlebarTitleVisibility.Hidden;
+            }
+#endif
+
+#if IOS
+            var statusBarStyle = AppInfo.Current.RequestedTheme == AppTheme.Dark ? UIKit.UIStatusBarStyle.LightContent : UIKit.UIStatusBarStyle.DarkContent;
+            Device.BeginInvokeOnMainThread(() =>
```

Comment:
Use `MainThread.BeginInvokeAsync`
Also put the whole method and this lamba into try/catch that passes exp to ExceptionHandler